File: Phind-CodeLlama-34B-v2/context-template-full/declare_directive_device_resident_clause.f90
#ifndef T1
!T1:declare directive device_resident clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(8), ALLOCATABLE, DEVICE_RESIDENT :: a(:)
        INTEGER :: n = 1000

        ALLOCATE(a(n))
        a = 1.0

        !$acc parallel loop device_resident(a)
        DO i = 1, n
          a(i) = a(i) + 1.0
        END DO
        !$acc end parallel loop

        DO i = 1, n
          IF (a(i) .ne. 2.0) THEN
            errors = errors + 1
          END IF
        END DO

        DEALLOCATE(a)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END FUNCTION test1
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: Phind-CodeLlama-34B-v2/context-template-full/declare_directive_device_resident_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
/*T1:declare directive device_resident clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    srand(SEED);

    int *a;
    int n = 100;

    #pragma acc enter data copyin(n)
    #pragma acc declare device_resident(a)

    a = (int*)malloc(n * sizeof(int));

    if (a == NULL){
        err = 1;
    }

    #pragma acc exit data delete(a)
    free(a);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}